pkg/services/accesscontrol/middleware.go

```
@@ -179,58 +180,107 @@ type userCache interface {
-					userCopy.OrgName = queryResult.OrgName
-					userCopy.OrgRole = queryResult.OrgRole
-				}
-			}
-
-			if userCopy.Permissions[targetOrgID] == nil {
-				permissions, err := service.GetUserPermissions(c.Req.Context(), &userCopy, Options{})
-				if err != nil {
-					deny(c, nil, fmt.Errorf("failed to authenticate user in target org: %w", err))
-				}
-
-				// guard against nil map
-				if userCopy.Permissions == nil {
-					userCopy.Permissions = make(map[int64]map[string][]string)
-				}
-				userCopy.Permissions[targetOrgID] = GroupScopesByAction(permissions)
+			tmpUser, err := makeTmpUser(c.Req.Context(), service, userService, teamService, c.SignedInUser, targetOrgID)
+			if err != nil {
+				deny(c, nil, fmt.Errorf("failed to authenticate user in target org: %w", err))
+				return
 			}
 
-			authorize(c, ac, &userCopy, evaluator)
+			authorize(c, ac, tmpUser, evaluator)
 
 			// guard against nil map
 			if c.SignedInUser.Permissions == nil {
 				c.SignedInUser.Permissions = make(map[int64]map[string][]string)
 			}
-			c.SignedInUser.Permissions[targetOrgID] = userCopy.Permissions[targetOrgID]
+			c.SignedInUser.Permissions[tmpUser.GetOrgID()] = tmpUser.GetPermissions()
+		}
+	}
+}
+
+// makeTmpUser creates a temporary user that can be used to evaluate access across orgs.
+func makeTmpUser(ctx context.Context, service Service, cache userCache,
+	teamService teamService, reqUser identity.Requester, targetOrgID int64) (identity.Requester, error) {
+	tmpUser := &user.SignedInUser{
+		OrgID:          reqUser.GetOrgID(),
+		OrgName:        reqUser.GetOrgName(),
+		OrgRole:        reqUser.GetOrgRole(),
+		IsGrafanaAdmin: reqUser.GetIsGrafanaAdmin(),
+		Login:          reqUser.GetLogin(),
+		Teams:          reqUser.GetTeams(),
+		Permissions: map[int64]map[string][]string{
+			reqUser.GetOrgID(): reqUser.GetPermissions(),
+		},
+	}
+
+	namespace, identifier := reqUser.GetNamespacedID()
+	id, _ := identity.IntIdentifier(namespace, identifier)
```

Why ignoring the error there?